src/Compilers/CSharp/Test/Symbol/Symbols/ExtendedPartialMethodsTests.cs

```diff
@@ -0,0 +1,2018 @@
+        }
+
+        [Fact]
+        public void Override_Mismatch_01()
+        {
+            const string text1 = @"
+class C
+{
+    internal virtual void M1() { }
+}
+partial class D : C
+{
+    internal override partial void M1();
+    internal partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (9,27): error CS8798: Both partial method declarations must have equal combinations of 'virtual', 'override', 'sealed', or 'new' modifiers.
+                //     internal partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodExtendedModDifference, "M1").WithLocation(9, 27)
+            );
+        }
+
+        [Fact]
+        public void Override_Mismatch_02()
+        {
+            const string text1 = @"
+class C
+{
+    internal virtual void M1() { }
+}
+partial class D : C
+{
+    internal partial void M1();
+    internal override partial void M1() { }
+}";
+            var comp = CreateCompilation(text1, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (8,27): warning CS0114: 'D.M1()' hides inherited member 'C.M1()'. To make the current member override that implementation, add the override keyword. Otherwise add the new keyword.
+                //     internal partial void M1();
+                Diagnostic(ErrorCode.WRN_NewOrOverrideExpected, "M1").WithArguments("D.M1()", "C.M1()").WithLocation(8, 27),
+                // (9,36): error CS8798: Both partial method declarations must have equal combinations of 'virtual', 'override', 'sealed', or 'new' modifiers.
+                //     internal override partial void M1() { }
+                Diagnostic(ErrorCode.ERR_PartialMethodExtendedModDifference, "M1").WithLocation(9, 36)
+            );
+        }
+
+        [Fact]
+        public void Override_AllowNull_01()
+        {
+            const string text1 = @"
+#nullable enable
+using System.Diagnostics.CodeAnalysis;
+
+class C
+{
+    internal virtual void M1([AllowNull] string s) { }
+}
+partial class D : C
+{
+    internal override partial void M1(string s1); // 1
+    internal override partial void M1(string s1) { }
+}";
+            var comp = CreateCompilation(new[] { text1, AllowNullAttributeDefinition }, parseOptions: TestOptions.RegularWithExtendedPartialMethods);
+            comp.VerifyDiagnostics(
+                // (11,36): warning CS8765: Nullability of type of parameter 's1' doesn't match overridden member (possibly because of nullability attributes).
+                //     internal override partial void M1(string s1); // 1
+                Diagnostic(ErrorCode.WRN_TopLevelNullabilityMismatchInParameterTypeOnOverride, "M1").WithArguments("s1").WithLocation(11, 36)
+            );
+        }
+
+        [Fact]
+        public void Override_AllowNull_02()
```

Comment:
Where is there a test where we have a `override partial` which involves `[AllowNull]` that succeeds?

Reply:
I'm not sure I understand. `Override_AllowNull_02` has the expected combination of `[AllowNull]` given on the parameters of the override, and it warns on dereference of the parameters. That seems successful.